### DISCUSSION TITLE
How to modify P2P IP addressing ?

### DISCUSSION THREAD

**Discussion by jmd1** [aristanetworks/avd#4016] [General]
Hi Team, 

I need guidance how to modify p2p IP addressing in AVD

I have a /24 for P2P between the SPINE and LEAFS links:

SPINE1 to Leaf - x.x.x.0 /26
SPINE2 to Leaf - x.x.x.64 /26
SPINE3 to Leaf - x.x.x.128 /26
SPINE4 to Leaf - x.x.x.192 /26

Thank you!

**Comment by ClausHolbechArista** [comment]
Hi James

There are multiple ways (ordered by my preference):

1. Use the new data model `downlink_pools` set per spine. It requires you to specify a range of interfaces which will be used to index into the pool. It *must* be set per spine, which also matches your use case. This is not available in any shipping AVD version, but will be available in AVD 4.8.0. Until then you can install `devel`. See the [installation guide](https://avd.sh/en/devel/docs/installation/collection-installation.html#install-latest-devel-version-from-avd-github) for how to install `devel`. See the `devel` docs for details on the new model: https://avd.sh/en/devel/roles/eos_designs/docs/input-variables.html#node-type-uplink-management (search for `downlink_pools`).

2. Override the default logic to build point-to-point IPs. See https://avd.sh/en/devel/roles/eos_designs/docs/input-variables.html#node-type-customization for details on how to set the custom templates.

3. Override the builtin python code to build point-to-point IPs. See same docs as 2.
